code/game/say.dm

```diff
@@ -111,24 +118,29 @@ GLOBAL_LIST_INIT(freqtospan, list(
 	if(copytext_char(input, -2) == "!!")
 		spans |= SPAN_YELL
 
+	var/say_mod = message_mods[MODE_CUSTOM_SAY_EMOTE]
+	if(!say_mod)
+		say_mod = say_mod(input, message_mods)
+
 	var/spanned = attach_spans(input, spans)
-	return "[say_mod(input, message_mods)], \"[spanned]\""
+	return "[say_mod], \"[spanned]\""
+
+/atom/proc/say_emphasis(message, var/list/ignore = list())
+	var/regex/markup
```

Comment:
can you document what the hell this proc does, because I do not understand by looking at it

Reply:
Done to the best of my current knowledge of it